### DISCUSSION TITLE
Screenshots not showing up in App Store Connect after running `upload_to_app_store` with an output of "Successfully uploaded screenshots to App Store Connect"

### DISCUSSION THREAD

**Discussion by mariabernis-tw** [fastlane/fastlane#16851] [Q&A] [answered]
### Description
When running action `upload_to_app_store`, it finishes successfully, says it has uploaded the screenshots but we don't get those images in the App Store Connect portal. 

We see the updated metadata and all other info of the release, but no screenshots: 
<img width="1256" alt="Screenshot 2020-07-14 at 16 37 53" src="https://user-images.githubusercontent.com/63052136/87438991-8278d000-c5f0-11ea-924b-5af83f83d453.png">

Nothing inside the "Media Manager" either.  

"Promotional Text", "Keywords", etc, also the "App Review Information" gets updated, but not the screenshots.

We got the screenshots in the App Store Connect portal the first time we ran `upload_to_app_store` for a build X. Following time we ran it (for build X+1), it removed the screenshots (we have `overwrite_screenshots` set to `true`) and despite the output that the screenshots were uploaded, they are nowhere to be seen. 😳 

##### Command executed
```ruby
platform :ios do
  desc "Promotes the latest iOS version in test flight for review (it doesn't publish automatically after review)"
  lane :deploy_prod do
    ENV['FASTLANE_PASSWORD'] = ios_apple_id_password
    release_notes_en = load_changelog(:ios, 'en-GB')
    upload_to_app_store(
        username: ios_apple_id_username,
        team_id: ios_app_store_connect_team_id,
        app_identifier: app_identifier,
        app_icon: ios_app_icon_path,
        price_tier: 0,
        submit_for_review: false, 
        reject_if_possible: false,
        force: true,
        automatic_release: false,
        overwrite_screenshots: true,
        skip_binary_upload: true,
        ignore_language_directory_validation: true,
        metadata_path: ios_metadata_path,
        screenshots_path: ios_screenshots_path,
        app_rating_config_path: ios_app_rating_path,
        release_notes: {
            'default' => release_notes_en,
            'en-GB' => release_notes_en
        },
        app_review_information: {
            email_address: ios_review_info["contact_email"],
            first_name: ios_review_info["contact_first_name"],
            last_name: ios_review_info["contact_last_name"],
            phone_number: ios_review_info["contact_phone"],
            notes: ios_review_info["notes"],
        })
  end
end
```

##### This is the output when running fastlane ios deploy_prod
```
[16:27:09]: Starting with the upload of screenshots...
[✔] Activating language en-GB...
[16:27:11]: Uploading 8 screenshots for language en-GB
[16:27:11]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/1_iphone55.ios-screenshot.png'...
[16:27:14]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/1_iphone65.ios-screenshot.png'...
[16:27:22]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/2_iphone55.ios-screenshot.png'...
[16:27:28]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/2_iphone65.ios-screenshot.png'...
[16:27:32]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/3_iphone55.ios-screenshot.png'...
[16:27:37]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/3_iphone65.ios-screenshot.png'...
[16:27:44]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/4_iphone55.ios-screenshot.png'...
[16:27:46]: Uploading '/path/to/my/repo/fastlane/metadata/ios/en-GB/4_iphone65.ios-screenshot.png'...
[✔] Saving changes
[16:28:03]: Successfully uploaded screenshots to App Store Connect
```

**Comment by joshdholtz** [comment, marked as answer]
@mariabernis-tw That is really strange! Can you try setting this environment variable before calling `upload_to_app_store`?

```
ENV["DELIVER_SKIP_WAIT_FOR_SCREENSHOT_PROCESSING"] = false
```

This will force the waiting for screenshot to process (which can take some time) but it might show if there are any errors that are causing them to not appear.

You could also try to upload a screenshot or two directly with App Store Connect website to see if there are any errors uploading there.

If you could get back to me on either or both of these that would be 💯

**Reply by mariabernis-tw** [reply]
Hi @joshdholtz! Thanks for the quick response! 
Latest fastlane version works perfectly, we were using 2.148.1. 
And the output now is much more informative, thanks for that :) 

PS: I've tried the env flag set to both false and true and it works 👌 . Also uploading pics from the App Store Connect portal.
